eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageTypes.scala

```diff
@@ -171,6 +171,51 @@ case class AcceptChannel(temporaryChannelId: ByteVector32,
   val channelType_opt: Option[ChannelType] = tlvStream.get[ChannelTlv.ChannelTypeTlv].map(_.channelType)
 }
 
+// NB: this message is named is open_channel2 in the specification.
+case class OpenDualFundedChannel(chainHash: ByteVector32,
+                                 temporaryChannelId: ByteVector32,
+                                 fundingFeerate: FeeratePerKw,
+                                 commitmentFeerate: FeeratePerKw,
+                                 fundingAmount: Satoshi,
+                                 dustLimit: Satoshi,
+                                 maxHtlcValueInFlightMsat: UInt64, // this is not MilliSatoshi because it can exceed the total amount of MilliSatoshi
+                                 htlcMinimum: MilliSatoshi,
+                                 toSelfDelay: CltvExpiryDelta,
+                                 maxAcceptedHtlcs: Int,
+                                 lockTime: Long,
+                                 fundingPubkey: PublicKey,
+                                 revocationBasepoint: PublicKey,
+                                 paymentBasepoint: PublicKey,
+                                 delayedPaymentBasepoint: PublicKey,
+                                 htlcBasepoint: PublicKey,
+                                 firstPerCommitmentPoint: PublicKey,
+                                 tlvStream: TlvStream[OpenDualFundedChannelTlv] = TlvStream.empty) extends ChannelMessage with HasTemporaryChannelId with HasChainHash {
+  val upfrontShutdownScript_opt: Option[ByteVector] = tlvStream.get[ChannelTlv.UpfrontShutdownScriptTlv].map(_.script)
+  val channelType_opt: Option[ChannelType] = tlvStream.get[ChannelTlv.ChannelTypeTlv].map(_.channelType)
+  val announceChannel: Boolean = tlvStream.get[ChannelTlv.DualFundedChannelFlagsTlv].exists(_.announceChannel)
+}
+
+// NB: this message is named is accept_channel2 in the specification.
```

Comment:
nit:
// NB: this message is named open_channel2 in the specification.

Reply:
Thanks! Fixed in https://github.com/ACINQ/eclair/pull/2231/commits/ddb61854537ffad723002a444a18684dc58d4a6e